### PR TITLE
add swtpm demo

### PR DESCRIPTION
**Pull request by DNJha**
This PR:
- adds software trusted platform module (SWTPM) to run inside SGX
- test script for IBM's TPM2.0 TSS

Tested on Azure confidential Computing Environment (_Standard DC2ds v3_) with Ubuntu 20.04.

### CODE REVIEW

**Comment by jessehui** [comment]
@DNJha Thank you for your delicate work! This is a great contribution to the Occlum project. I took a quick look at this PR.  In general, this looks really good. There are just minor questions regarding this PR:
- If I understand correctly, the `demos/swtpm/run_swtpm.sh` only starts the server, but the client command is not included. Could you also add a client command to make this test script complete?
- With the complete script, could you also add this demo to our CI? Most of the demos have their CI defined at `.github/workflows/demo_test.yml`. This can help us make sure the demos work for new Occlum versions.
- Please also include SWTPM at demos/README.md with just a simple line

Thank you!

**Review by jessehui — COMMENTED** [review]
Thank you for your delicate work! This is a great contribution to the Occlum project. I took a quick look at this PR. In general, this looks really good. There are just minor questions regarding this PR:

- If I understand correctly, the demos/swtpm/run_swtpm.sh only starts the server, but the client command is not included. Could you also add a client command to make this test script complete?

- With the complete script, could you also add this demo to our CI? Most of the demos have their CI defined at .github/workflows/demo_test.yml. This can help us make sure the demos work for new Occlum versions.

- Please also include SWTPM at demos/README.md with just a simple line

Thank you!

**Comment by DNJha** [comment]
As suggested, I added IBM's TPM2.0 TSS client to test the functioning of SWTPM, updated the README file and defined the script for CI.

**Comment by jessehui** [comment]
@DNJha Thank you for updating! But it looks like the CI failed. I think it is due to the formatting issue of the `swtpm.yaml`. Please take a look. After fixing this, please squash all the commits into a single one. And you can force push to your branch of this PR. Thank you.

I have another question when running your demo. When running under my server, it looks like the `make check` command in the `install_swtpm.sh` failed. I checked the CI status and it looks like with GitHub's CI machine, all checks passed. Do you know why and how can I fix this?
截屏2023-07-31 09 53 49

**Comment by DNJha** [comment]
@jessehui I have updated the `swtpm.yaml` file as well as the `install_swtpm.sh` to replace wget with a git clone. I also squashed the commits to one commit as suggested. 

The `make check ` was failing because of the update on `libtpms`. I added the dependencies to resolve that, I hope this will work now. Please let me know if it still fails.

File: demos/swtpm/run_swtpm.sh [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+# Build Occlum
+echo -e "${BLUE}Occlum build swtpm${NC}"
+occlum build
+
+# Run the python demo
+echo -e "${BLUE}Occlum start swtpm${NC}"
+
+occlum run /bin/swtpm socket --tpmstate dir=/bin/myvtpm --tpm2 --ctrl type=tcp,port=2322,bindaddr=0.0.0.0 --server type=tcp,port=2321,bindaddr=0.0.0.0 
+--flags not-need-init --seccomp action=none
```
Comment (jessehui) [inline demos/swtpm/run_swtpm.sh:24]: This script can't run correctly. This command should be in one line.

File: .github/workflows/demo_test.yml [hunk 2]
```diff
@@ -871,3 +871,24 @@ jobs:
...
+    - uses: ./.github/workflows/composite_action/sim
+      with:
+        container-name: ${{ github.job }}
+        build-envs: 'OCCLUM_RELEASE_BUILD=1'
+
+    - name: download and build swtpm
+      run: docker exec ${{ github.job }} bash -c "cd /root/occlum/demos/swtpm; ./install_swtpm.sh"
+
+    - name: Run swtpm server
+      run: docker exec ${{ github.job }} bash -c "cd /root/occlum/demos/swtpm; SGX_MODE=SIM ./run_swtpm.sh"
```
Comment (jessehui) [inline .github/workflows/demo_test.yml:891]: `run_swtpm.sh` will run as a server and this script will never return. You can take a look at the `Https_server_test` test in this file to use the background process to run this test.

**Comment by jessehui** [comment]
@DNJha Thanks for your update. However, with the latest code, I still got many failure cases when running the `install_swtpm.sh` script.  Do you have any idea why this happened?
截屏2023-08-23 17 00 44

This check error doesn't prevent me from running this demo. I give two extra comments. Please take a look. Thank you!

File: demos/swtpm/run_swtpm.sh (hunk 1 above)
Comment (DNJha) [reply demos/swtpm/run_swtpm.sh]: Sorry for the typo, I fixed it now

File: .github/workflows/demo_test.yml (hunk 2 above)
Comment (DNJha) [reply .github/workflows/demo_test.yml]: I updated it following the` Https_server_test`. Hope the CI will pass this time

**Review by jessehui — APPROVED** [review]
LGTM. Thank you for your contribution!

**Comment by jessehui** [comment]
@DNJha Thank you for your update. I think it's fine. Without passing all the checks, this demo can also run. And it seems the CI has passed. I think it is close to merging.

**Review by volcano0dr — APPROVED** [review]

**Comment by DNJha** [comment]
Thank you.

**Pull request merged**